### ISSUE TITLE
Logging to stdout (instead of stderr) is a wrong default

### ISSUE THREAD

**Issue by dpc** [tokio-rs/tracing#2492]
## Bug Report

I just lost decent amount of time debugging a problem that turned out to be `tracing-subscriber` logging to `stdout` which was overlooked during refactoring. This is a second, if not third time in the recent past where I've dealt with wrong defaults creating issues..

I don't recall ever wanting to get my log output to `stdout`, and as far as I can tell logging to stderr is a well estabilished convention, to separate main program output from diagnostics.

`env_logger` default to `stderr` https://github.com/rust-cli/env_logger/blob/7e150d9c9fc45812a684aedff65bb5a6a0ad21a0/README.md?plain=1#L166 so does most logging libraries I've used and can remember.

Is there any reason why `tokio-subscriber` would go against this convention?

**Comment by mkpankov** [comment]
Also surprised with this.
This might be a performance consideration since logging a lot of data to unbuffered sink might be less performant. But I'd like the team to clarify this.

**Comment by dpc** [comment]
stdout will flush on newlines (uses [std::io::LineWriter](https://doc.rust-lang.org/std/io/struct.LineWriter.html#) under the hood). Should be natural for any logging library to just write-out stuff in lines (or use `LineWriter` to gather output into newlines internally). In addition `stdout` is locked, which is making it possibly slower. So I doubt it is any perf optimization.

**Comment by hawkw** [comment]
I agree that it would probably be more correct to make stderr the default. We should probably change it.

IMO, changing the output stream is semver-breaking, so we would need to do this in a `tracing-subscriber` 0.4. If we do an 0.4 release soon, we probably want to reassess some other default behaviors as well.

**Comment by Natr1x** [comment]
> I agree that it would probably be more correct to make stderr the default. We should probably change it.
> 
> IMO, changing the default output stream is semver-breaking, so we would need to do this in a `tracing-subscriber` 0.4. If we do an 0.4 release soon, we probably want to reassess some other default behaviors as well.

Is it possible to put a warning/clarification in the documentation and README examples in the meantime. I also just lost a couple of hours due to this.

**Comment by Victor-N-Suadicani** [comment]
Also just ran into this. The `0.3` version is from 2021 - perhaps the time is ripe soon for a breaking `0.4` release that could fix this? :)

**Comment by Infinoid** [comment]
@Victor-N-Suadicani thanks!  Sadly, searching the docs for "stderr" [misses](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/index.html?search=stderr) that, only finding results related to unit testing.
